lib/sensors/src/URM04Sensor.cpp

```diff
@@ -0,0 +1,216 @@
+#include "URM04Sensor.h"
+
+// instantiate pin connected to the URM04 URM04Sensor
+URM04Sensor::URM04Sensor::URM04Sensor(PinName trig_pin, PinName _RX, PinName _TX, uint8_t default_address)
+    : m_trigPin(trig_pin), startAddr(default_address), RX(_RX), TX(_TX), serial(TX, RX, BAUD_RATE) {
+  // instantiate command buffer with all zeros
+
+  /***
+   * for(int i{0}; i<10; i++){
+   *   cmdst[i] = 0;
+   * }
+   *
+   ***/
+  memset(&cmdst[0], 0, sizeof(cmdst));
+}
+
+void URM04Sensor::URM04Sensor::trigger_sensor(float& distance) {
```

Comment:
Since the trigger command just triggers the sensor to make a measurement but doesn't actually transmit the measurement, it would make more sense to remove the `distance` param and just let the function return a `bool` to indicate if the trigger was successful